src/main/kotlin/io/snyk/plugin/SnykBulkFileListener.kt

```diff
@@ -79,9 +85,29 @@ class SnykBulkFileListener : BulkFileListener {
                     .filter { iacFileExtensions.contains(it.extension) }
                     .find { ProjectRootManager.getInstance(project).fileIndex.isInContent(it) }
                 if (iacRelatedFileChanged != null) {
-                    toolWindowPanel.currentIacResult = null
-                    LOG.debug("IaC cached results dropped due to changes in: $iacRelatedFileChanged")
-                    DaemonCodeAnalyzer.getInstance(project).restart()
+                    val iacTreeRoot = toolWindowPanel.getRootIacIssuesTreeNode()
+                    val nodes = iacTreeRoot.depthFirstEnumeration()
+                    while (nodes.hasMoreElements()) {
+                        val node = nodes.nextElement() as DefaultMutableTreeNode
+                        if (node is IacFileTreeNode) {
+                            val iacIssuesForFile = node.userObject as IacIssuesForFile
+                            val iacFile = VirtualFileManager.getInstance().findFileByNioPath(
+                                Paths.get(iacIssuesForFile.targetFilePath)
+                            )
+                            if (iacFile != null && virtualFilesAffected.contains(iacFile)) {
+                                val obsolete = " (obsolete)"
+                                if (!iacIssuesForFile.targetFile.endsWith(obsolete)) {
+                                    iacIssuesForFile.targetFile += obsolete
+                                }
```

Comment:
we shouldn't change data fields gotten from json. In that particular case navigation will be broken. to change display presentation we use either in VulnerabilityTreeCellRenderer or SnykToolWindowPanel#updateTreeRootNodesPresentation